### PR TITLE
chore(traffic-guards): more details in error message and log

### PR DESCRIPTION
**Pull request by dreynaud**
Would be helpful to know what _has_ been found when the error is "could
not find x".

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java [hunk 1]
```diff
@@ -125,15 +125,15 @@ public void verifyTrafficRemoval(String serverGroupName, Moniker serverGroupMoni
     List<TargetServerGroup> targetServerGroups = ((List<Map<String, Object>>) cluster.get().get("serverGroups"))
       .stream()
       .map(TargetServerGroup::new)
-      .filter(tsg -> location.equals(tsg.getLocation()))
+      .filter(tsg -> location.equals(tsg.getLocation(location.getType())))
```
Comment (dreynaud) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java]: @ewiseblatt this is the fix attempt we were discussing earlier

**Comment by ewiseblatt** [comment]
This lets the kube_smoke_test pass once again. I'm not sure if you want the added log of the non-matching services. I'm not sure if that's a common case or not.

I dont really understand the fix. I see you took away the precondition assertion with a softer check and return, but it isnt clear what's going on in this case as to where the problem is and how to avoid reintroducing it with a future change somewhere.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java (hunk 1 above)
Comment (ewiseblatt) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java]: ok. thanks for pointing that out. I thought it was a different one.

**Comment by ewiseblatt** [comment]
LGTM. I appreciate your help looking into this.

**Pull request merged**